components-api/src/main/scala/pl/touk/nussknacker/engine/api/context/ValidationContext.scala

```diff
@@ -13,6 +13,17 @@ import javax.lang.model.SourceVersion
 object ValidationContext {
 
   def empty: ValidationContext = ValidationContext()
+
+  def isVariableNameValid(name: String): Boolean = SourceVersion.isIdentifier(name)
+
+  def validateVariableName(name: String, paramName: Option[String])(
+      implicit nodeId: NodeId
+  ): ValidatedNel[PartSubGraphCompilationError, String] = {
+    // TODO: add correct and more precise error messages
+    if (isVariableNameValid(name)) Valid(name)
+    else Invalid(InvalidVariableOutputName(name, paramName)).toValidatedNel
```

Comment:
i think `InvalidVariableOutputName` is not a perfect name, it's should just be `InvalidVariableName`, the 'output' part looks like a legacy leftover

Reply:
its text in PrettyValidationError.scala also should change

Reply:
I didn't notice that, I think you're right. I've changed it and also message in OverwrittenVariable error.

Reply:
FYI, the error message from InvalidVariableName is not correct since the we use normal java variable identifier rules and they are not as simple as `only letters, numbers or '_', cannot be empty` :smile: I've left a TODO to provide more precise error messages.